### ISSUE TITLE
Loading resources over HTTP should produce informative, consistent exceptions

### ISSUE THREAD

**Issue by darraghjburke** [BabylonJS/BabylonNative#759]
Issue: we don't currently know what passing an invalid URL to XMLHTTPRequest will do across different platforms, and we're not sure what the correct behavior should be. We probably want to throw a JavaScript exception, but should that be thrown in the Send() method or the Open() method?

Things to consider:

- How closely should we match behavior on the web? [The spec does not explicitly specify where exceptions should be thrown](https://xhr.spec.whatwg.org/#the-open()-method), but typically browsers will throw DOMExceptions on open() if the path is invalid.
- Is it even possible to throw exceptions from open() on all platforms? That would require that the platform allows us to parse the URL as a separate action from the network request, which may not be the case for all platforms. (it is at least true for Android)
- What are we currently doing on Windows and iOS?

The solution here is to decide on a contract and ensure that it's implemented consistently across all platforms. It would be nice to have unit tests for these exceptions as well.

Feel free to add any further comments @bghgary

**Comment by darraghjburke** [comment]
As of right now, the behavior on Windows is similar to the (unpatched) Android behavior. Attempting to load "noprotocol.gltf" or similar causes the following error:

> Exception thrown at 0x00007FFDCBA54B89 (KernelBase.dll) in Playground.exe: WinRT originate error - 0x80070057 : 'noprotocol.gltf is not a valid absolute URI.'.

This leads to the program aborting. We may want to make this behavior match the new Android behavior, where the exception is passed to the Javascript code.

**Comment by darraghjburke** [comment]
Waiting on #802, want to make sure that the code has strong unit testing in place before making further changes in order to prevent regressions.

**Comment by CedricGuillemet** [comment]
Got an interesting issue while doing a Validation tests run on iOS device:
```
ValidationTests[9302:12667787] App Transport Security has blocked a cleartext HTTP (http://) resource load since it is insecure. Temporary exceptions can be configured via your app's Info.plist file.
2021-07-02 14:52:23.013913+0200 ValidationTests[9302:12667787] Cannot start load of Task <CBB9B7B6-C19A-494A-88EE-C6511FDD3F6A>.<28> since it does not conform to ATS policy
2021-07-02 14:52:23.018328+0200 ValidationTests[9302:12667787] Task <CBB9B7B6-C19A-494A-88EE-C6511FDD3F6A>.<28> finished with error [-1022] Error Domain=NSURLErrorDomain Code=-1022 "The resource could not be loaded because the App Transport Security policy requires the use of a secure connection." UserInfo={NSLocalizedDescription=The resource could not be loaded because the App Transport Security policy requires the use of a secure connection., NSErrorFailingURLStringKey=http://models.babylonjs.com/Tests/TextureTransform/TextureTransformTest.gltf, NSErrorFailingURLKey=http://models.babylonjs.com/Tests/TextureTransform/TextureTransformTest.gltf, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <CBB9B7B6-C19A-494A-88EE-C6511FDD3F6A>.<28>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <CBB9B7B6-C19A-494A-88EE-C6511FDD3F6A>.<28>, NSUnderlyingError=0x28180a250 {Error Domain=kCFErrorDomainCFNetwork Code=-1022 "(null)"}}
```
I didn't go any further with that. But it might be an interesting test to check https and http work on all platforms.
